pkg/services/queryhistory/api.go

```diff
@@ -52,3 +61,37 @@ func (s *QueryHistoryService) deleteHandler(c *models.ReqContext) response.Respo
 		ID:      id,
 	})
 }
+
+func (s *QueryHistoryService) starHandler(c *models.ReqContext) response.Response {
+	queryUID := web.Params(c.Req)[":uid"]
+	if len(queryUID) == 0 || !util.IsValidShortUID(queryUID) {
+		return response.Error(http.StatusNotFound, "Query in query history not found", nil)
+	}
+
+	id, err := s.StarQueryInQueryHistory(c.Req.Context(), c.SignedInUser, queryUID)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to star query in query history", err)
+	}
+
+	return response.JSON(http.StatusOK, StarQueryInQueryHistoryResponse{
```

Comment:
Could we return here a DTO as in `patchQueryComment`? We need to get the DTO from the table anyway before starring so it shouldn't be a problem. Getting full DTO could help keeping UI and DB in sync especially once multiple users edit the same query.

Reply:
Yes we can! I was thinking about it, but decided not to do that. But will do!